cunumeric/module.py

```diff
@@ -6227,3 +6227,701 @@ def bincount(
+
+
+# Weibull method:
+# 'weibull'
+#
+def weibull(q, n):
+    pos = q * (n + 1)
+
+    k = math.floor(pos)
+    # gamma = pos-k
+    #
+    # this fixes it:
+    # (when pos > n: this actually selects the right point,
+    #  which is the correct choice, because right = arr[n]
+    #  gets invalidated)
+    #
+    gamma = 0.0 if (pos < 1 or pos > n) else pos - k
+
+    j = floor_i(k)
+    return (gamma, j)
+
+
+# Gumbel method:
+# 'linear'
+#
+def linear(q, n):
+    pos = q * (n - 1) + 1
+    k = math.floor(pos)
+    # gamma = pos-k
+    #
+    # this fixes it:
+    # (when pos > n: this actually selects the right point,
+    #  which is the correct choice, because right = arr[n]
+    #  gets invalidated)
+    #
+    gamma = 0.0 if (pos < 1 or pos > n) else pos - k
+
+    j = floor_i(k)
+    return (gamma, j)
+
+
+# Johnson & Kotz method:
+# 'median_unbiased'
+#
+def median_unbiased(q, n):
+    fract = 1.0 / 3.0
+    pos = q * (n + fract) + fract
+    k = math.floor(pos)
+
+    # gamma = pos-k
+    #
+    # this fixes it:
+    # (when pos > n: this actually selects the right point,
+    #  which is the correct choice, because right = arr[n]
+    #  gets invalidated)
+    #
+    gamma = 0.0 if (pos < 1 or pos > n) else pos - k
+
+    j = floor_i(k)
+    return (gamma, j)
+
+
+# Blom method:
+# 'normal_unbiased'
+#
+def normal_unbiased(q, n):
+    fract1 = 0.25
+    fract2 = 3.0 / 8.0
+    pos = q * (n + fract1) + fract2
+    k = math.floor(pos)
+
+    # gamma = pos-k
+    #
+    # this fixes it:
+    # (when pos > n: this actually selects the right point,
+    #  which is the correct choice, because right = arr[n]
+    #  gets invalidated)
+    #
+    gamma = 0.0 if (pos < 1 or pos > n) else pos - k
+
+    j = floor_i(k)
+    return (gamma, j)
+
+
+# `lower`
+#
+def lower(q, n):
+    gamma = 0.0
+    pos = q * (n - 1)
+    k = math.floor(pos)
+
+    j = int(k)
+    return (gamma, j)
+
+
+# `higher`
+#
+def higher(q, n):
+    pos = q * (n - 1)
+    k = math.floor(pos)
+
+    # Generic rule: (k == pos)
+    gamma = 0.0 if (pos == 0 or k == pos) else 1.0
+
+    j = int(k)
+    return (gamma, j)
+
+
+# `midpoint`
+#
+def midpoint(q, n):
+    pos = q * (n - 1)
+    k = math.floor(pos)
+
+    # Generic rule: (k == pos)
+    gamma = 0.0 if (pos == 0 or k == pos) else 0.5
+
+    j = int(k)
+    return (gamma, j)
+
+
+# `nearest`
+#
+def nearest(q, n):
+    pos = q * (n - 1)
+
+    # k = floor(pos)
+    # gamma = 1.0 if pos - k >= 0.5 else 0.0
+
+    k = np.round(pos)
+    gamma = 0.0
+
+    j = int(k)
+    return (gamma, j)
+
+
+# helper: test if array inputs are different:
+#
+def is_diff(arr1, arr2):
+    return len(arr1) != len(arr2) or any(
+        [x != y for (x, y) in zip(arr1, arr2)]
+    )
+
+
+# helper: in-place interpolator:
+# quantile interpolation method to act on vectors,
+# like an axpy call:
+# [gamma](vleft, vright) { return (1-gamma)*vleft + gamma*vright;}
+#
+def in_place_interp(gamma, arr_lvals, arr_rvals, result):
```

Comment:
This needs to be "inlined" where it's called, otherwise this case doesn't work:
```
print(cn.quantile(cn.random.random((3,4,5)), [0.25, 0.75]))
```
There should be a test covering this case.

Reply:
Test added, but I couldn't repro the failure (with `in_place_interp()` as is).